### PR TITLE
Make sure the path is correctly escaped in case it contains spaces.

### LINKED ISSUE #1044: Can't run the specs from a path with spaces
When trying to run my specs I'm getting the following error: 

screen shot 2018-03-10 at 12 42 37 pm

The cause for the error is that my path where I store my project in has a space in it `.../Mobile Documents/...` For the bash command to work properly, the empty space needs to be escaped like so: `.../Mobile\ Documents/...`

Be happy to open up a PR to fix this!

### PR DESCRIPTION
**Pull request by andrewmarkle**
Closes #1044. This fixes an issue where the test runner fails if you have spaces in your path. 

This PR uses Regexp to escape the path before the command is run.

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1045)

### CODE REVIEW

File: lib/react_on_rails/utils.rb [hunk 1]
```diff
@@ -146,7 +146,7 @@ def required(arg_name)
     end
 
     def self.prepend_cd_node_modules_directory(cmd)
-      "cd #{ReactOnRails.configuration.node_modules_location} && #{cmd}"
+      "cd #{Regexp::escape(ReactOnRails.configuration.node_modules_location.to_s)} && #{cmd}"
```
Comment (justin808) [inline lib/react_on_rails/utils.rb]: Why is this the right solution for shell escaping?

Comment (justin808) [reply lib/react_on_rails/utils.rb]: What if put quotes around the value:

```ruby
"cd \"#{ReactOnRails.configuration.node_modules_location}\" && #{cmd}"
```

**Comment by justin808** [comment]
Travis failure:

```
Offenses:
lib/react_on_rails/utils.rb:149:19: C: Style/ColonMethodCall: Do not use :: for method calls.
      "cd #{Regexp::escape(ReactOnRails.configuration.node_modules_location.to_s)} && #{cmd}"
                  ^^
139 files inspected, 1 offense detected
rake aborted!
```

File: lib/react_on_rails/utils.rb (hunk 1 above)
Comment (andrewmarkle) [reply lib/react_on_rails/utils.rb]: Good idea! Didn't think of that. I like that solution better!

**Comment by justin808** [comment]
@andrewmarkle  we have this CI failure:

```
Failures:
  1) ReactOnRails::TestHelper::WebpackAssetsCompiler#ensureAssetsCompiled when assets compiler command is invalid prints the correct message
     Failure/Error:
       expect do
         begin
           ReactOnRails::TestHelper::WebpackAssetsCompiler.new.compile_assets
           # rubocop:disable Lint/HandleExceptions
         rescue SystemExit
           # No op
         end
         # rubocop:enable Lint/HandleExceptions
       end.to output(/#{expected_output}/).to_stdout
     
       expected block to output /React on Rails FATAL ERROR!
       Error in building webpack assets!
       cmd: cd  && sh -c 'exit 1'
       / to stdout, but output "\nBuilding Webpack assets...\n\e[31m================================================================...tatus: 1\n================================================================================\n\e[0m\n"
       Diff:
       
       @@ -1,5 +1,10 @@
       -/React on Rails FATAL ERROR!
       +
       +Building Webpack assets...
       +================================================================================
       +React on Rails FATAL ERROR!
        Error in building webpack assets!
       -cmd: cd  && sh -c 'exit 1'
       -/
       +cmd: cd "" && sh -c 'exit 1'
       +exitstatus: 1
       +================================================================================
       +
       
     # ./spec/react_on_rails/test_helper/webpack_assets_compiler_spec.rb:30:in `block (4 levels) in <top (required)>'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:115:in `block in run'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:104:in `loop'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:104:in `run'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rspec-retry-0.5.5/lib/rspec_ext/rspec_ext.rb:12:in `run_with_retry'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rspec-retry-0.5.5/lib/rspec/retry.rb:33:in `block (2 levels) in setup'
Finished in 1 minute 18.36 seconds (files took 1.29 seconds to load)
144 examples, 1 failure
Failed examples:
rspec ./spec/react_on_rails/test_helper/webpack_assets_compiler_spec.rb:23 # ReactOnRails::TestHelper::WebpackAssetsCompiler#ensureAssetsCompiled when assets compiler command is invalid prints the correct message
```

**Comment by justin808** [comment]
Reviewed 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[spec/react_on_rails/test_helper/webpack_assets_compiler_spec.rb, line 27 at r3](https://reviewable.io:443/reviews/shakacode/react_on_rails/1045#-L8GsobTj-hFW01lZIob:-L8GsobTj-hFW01lZIoc:b-qnaz3s) ([raw file](https://github.com/shakacode/react_on_rails/blob/144ecbaffb764090c70f15912f775cd981b964d9/spec/react_on_rails/test_helper/webpack_assets_compiler_spec.rb#L27)):*
> ```Ruby
>           React on Rails FATAL ERROR!
>           Error in building webpack assets!
>           cmd: cd \"#{Rails.root}\" && #{invalid_command}
> ```

If this is in a heredoc, why do you escape the `"`? @andrewmarkle 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/1045)*

**Comment by justin808** [comment]
Please add a changelog entry

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/1045#-:-L8GsxLJ__mikGGAkEEt:btd8eaf)*

**Comment by justin808** [comment]
@andrewmarkle ?

**Comment by andrewmarkle** [comment]
> If this is in a heredoc, why do you escape the "? 
I couldn't get this to work otherwise. If you just add in the escape characters, the heredoc ignores the interpolation and it outputs #{Rails.root} as part of the string. 

Do you have another suggestion?

**Comment by justin808** [comment]
LGTM

---

Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/shakacode/react_on_rails/1045#-:-L8yROCJCdTsoPplZSBa:bb6ywmx)*

**Pull request merged**